codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -104,12 +104,13 @@
     public static final int DEFAULT_WINDOW_SIZE = 65535;
     public static final short DEFAULT_PRIORITY_WEIGHT = 16;
     public static final int DEFAULT_HEADER_TABLE_SIZE = 4096;
+    public static final long INITIAL_SETTINGS_HEADER_LIST_SIZE = 8192;
```

Comment:
For a server, having this always hard coded at 8k leaves us open to clients configured with a different initial value crashing each new connection with an initial headers frame (before seeing settings from the server) large enough to trigger a GO_AWAY, potentially triggering an infinite loop of connection churn.

It seems this should be configurable so it can match whatever the client is configured with?